tests/unit/s2n_ktls_test_utils_test.c

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+#include "testlib/s2n_testlib.h"
+#include "utils/s2n_random.h"
+
+#define S2N_TEST_RECORD_TYPE 43
+
+/* Creates an iovec and sets it on new msghdr. This operation is verbose
+ * and not super interesting so its captured in a macro to make the tests
+ * easier to read. */
+#define INIT_MSGHDR(name, buf, len)       \
+    struct iovec name##_msg_iov = { 0 };  \
+    name##_msg_iov.iov_base = buf;        \
+    name##_msg_iov.iov_len = len;         \
+    struct msghdr name##_msg = { 0 };     \
+    name##_msg.msg_iov = &name##_msg_iov; \
+    name##_msg.msg_iovlen = 1;
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+
+    /* test data */
+    uint8_t test_data[S2N_TLS_MAXIMUM_FRAGMENT_LENGTH] = { 0 };
+    struct s2n_blob test_data_blob = { 0 };
+    EXPECT_SUCCESS(s2n_blob_init(&test_data_blob, test_data, sizeof(test_data)));
+    EXPECT_OK(s2n_get_public_random_data(&test_data_blob));
+
+    /* Test the test. Tests the mock IO stuffer implementation */
```

Comment:
I do think that this comment could be improved. Something along the lines of, "test the mocked sendmsg recvmsg io can successfully write and recv messages."